### PR TITLE
#calendar520 add delete method

### PR DESCRIPTION
**Pull request by asmaNode**

### CODE REVIEW

File: test/api/calendar.test.ts [hunk 1]
```diff
@@ -67,3 +67,36 @@ describe('the getInbox method', () => {
...
+    deleteEvent(davClient)(path);
+    expect(httpClient.requestText).toHaveBeenCalledWith({
+      url: 'http://url/calendars/events/123456.ics',
+      method: 'DELETE',
+      headers: {
+        Authorization: 'Basic header',
+      }
+    });
+  });
+});
```
Comment (renaudboyer) [inline test/api/calendar.test.ts:102]: Please add new line at the end of the file

File: test/api/calendar.test.ts [hunk 2]
```diff
@@ -67,3 +67,36 @@ describe('the getInbox method', () => {
...
+      headers: {
+        Authorization: 'Basic header',
+      },
+    });
+  });
+
+  it('should send a DELETE request with the event path as param', () => {
+    const path = 'events/123456.ics';
+    
+
```
Comment (renaudboyer) [inline test/api/calendar.test.ts:92]: Too many new lines

File: test/api/calendar.test.ts [hunk 3]
```diff
@@ -67,3 +67,36 @@ describe('the getInbox method', () => {
...
+    davClient = new DAVClient({
+      baseURL: 'http://url',
+      httpClient,
+      headers: {
+        Authorization: 'Basic header',
+      },
+    });
+  });
+
+  it('should send a DELETE request with the event path as param', () => {
```
Comment (rezk2ll) [inline test/api/calendar.test.ts:89]:
```suggestion
  it('should send a DELETE request to the event path', () => {
```

File: lib/api/calendars.ts [hunk 4]
```diff
@@ -42,3 +42,10 @@ export const getInbox = (client: DAVClient) => async (userId: string): Promise<C
     ),
   }));
 }
+
+export const deleteEvent = (client: DAVClient) => async (eventPath: string): Promise<string> => {
+  return await client.requestText({
```
Comment (rezk2ll) [inline lib/api/calendars.ts:47]:
```suggestion
  return await client.request({
```
no body was sent within the request, so no need to use `requestJson` or `requestText`

Comment (asmaNode) [reply lib/api/calendars.ts]: i must added it to the DAV client

**Comment by rezk2ll** [comment]
1 extra commit

**Review by renaudboyer — APPROVED** [review]

**Comment by asmaNode** [comment]
i have just rebased the code :/

File: lib/DAVClient.ts [hunk 5]
```diff
@@ -40,4 +40,7 @@ export class DAVClient {
   public requestText(options: RequestOptions): Promise<string> {
     return this.#httpClient.requestText(this.buildRequestOptions(options));
   }
+  public request(options: RequestOptions): Promise<Response> {
+    return this.#httpClient.request(this.buildRequestOptions(options));
+  }
```
Comment (huy-ta) [inline lib/DAVClient.ts:45]: - `requestText` is for when the response body is just text.
- `requestJson` is for when the response body is json.

In the case when the response body empty, I guess you can use `requestJson`, so no need to add this method.

File: lib/api/calendars.ts [hunk 6]
```diff
@@ -51,3 +51,10 @@ export const modifyEvent = (client: DAVClient) => async (eventPath: string, even
     body: translate(event)
   });
 }
+
+export const deleteEvent = (client: DAVClient) => async (eventPath: string): Promise<Response> => {
+  return await client.request({
+    url: `${BASE_PATH}/${eventPath}`,
```
Comment (huy-ta) [inline lib/api/calendars.ts:57]: Please use `url-join` (see Khaled's PRs).

**Comment by huy-ta** [comment]
> i have just rebased the code :/

Mhhh... That's a merge not a rebase.

File: lib/DAVClient.ts (hunk 5 above)
Comment (asmaNode) [reply lib/DAVClient.ts:45]: it was the review of @rezk2ll  to use request , in all cases we need to add this method , if not why it is added in HttpClient interface

Comment (huy-ta) [reply lib/DAVClient.ts]: You don't need to use a new method when there's already a method that can do the same job (which is `requestJson`).

**Review by rezk2ll — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]

**Pull request merged**